validator/src/main/java/com/google/daq/mqtt/registrar/Registrar.java

```diff
@@ -653,6 +679,23 @@ private Map<String, LocalDevice> loadAllDevices() {
     return loadDevices(SiteModel.listDevices(devicesDir));
   }
 
+  private Set<String> loadChangedDevices() {
+    if (optimizeRun && lastRunInstant != null) {
+      System.err.println("Collecting devices changed after " + lastRunInstant);
+      return allDevices.entrySet().stream()
+          .filter(entry -> {
+            LocalDevice device = entry.getValue();
+            Metadata metadata = device.getMetadata();
+            if (metadata == null || metadata.timestamp == null) {
```

Comment:
seems like you could combine this if with the following return since it's all boolean logic?